### ISSUE TITLE
ncbi_snp_query() breaks when snps contains rsid of SNP that is no longer supported by dbSNP

### ISSUE THREAD

**Issue by ellisifnygaard** [ropensci/rsnps#157]
**Describe the bug**
When `snps` includes an rsid of a SNP that is listed as "unsupported" by dbSNP ,  `ncbi_snp_query() ` breaks and does not return any data. This behaviour is problematic when the query contains multiple SNPs. In the source code, it appears that the function interprets all instances where `variant.response.content$primary_snapshot_data` does not exist as being due to the SNP being merged into another SNP: 
https://github.com/ropensci/rsnps/blob/4f434c04d78d68a356a2e373f46755d43a72cb8e/R/ncbi_snp_api.R#L275
  
```r
## if merged into another id
    if (is.null(variant.response.content$primary_snapshot_data)) {
      
      rsid <- as.character(paste0("rs", variant.response.content$merged_snapshot_data$merged_into))
      no_rsid <-  as.character(variant.response.content$merged_snapshot_data$merged_into)
      
      warning(Query, " has been merged into ", rsid, call. = FALSE)
      
      variant.url <- paste0("https://api.ncbi.nlm.nih.gov/variation/v0/refsnp/", no_rsid)
      variant.response <- httr::GET(variant.url)
      variant.response.content <- jsonlite::fromJSON(rawToChar(variant.response$content),
                                                     simplifyVector = FALSE)
      
    } else {
      rsid <- as.character(paste0("rs",
                                  variant.response.content$refsnp_id))
    }
```

This does not capture all situations where `is.null(variant.response.content$primary_snapshot_data)` is `TRUE`. When SNPs are not supported by dbSNP anymore, `variant.response.content$primary_snapshot_data` is `NULL`, but `variant.response.content$merged_snapshot_data` is also `NULL`, causing the code to break.

  
**Code to reproduce**

**rs10921627** is a SNP that is no longer supported by dbSNP: https://www.ncbi.nlm.nih.gov/snp/rs10921627

When this SNP is included in a query, alone or among other supported SNPs, the function breaks. The function and does not return information about other (supported) SNPs in `snps`.

```r
# Alone:
rsnps::ncbi_snp_query( "rs10921627" )
# Getting info about the following rsIDs: rs10921627
# Error: lexical error: invalid char in json text.
# <html>  <head><title>404 Not Fo
# (right here) ------^
#   In addition: Warning message:
#   rs10921627 has been merged into rs 

# Among other supported SNPs:
rsnps::ncbi_snp_query( c( "rs10921647", "rs10921627", "rs10921645" ) )
# Getting info about the following rsIDs: rs10921647, rs10921627, rs10921645
# Error: lexical error: invalid char in json text.
# <html>  <head><title>404 Not Fo
# (right here) ------^
#   In addition: Warning message:
#   rs10921627 has been merged into rs 

# When excluding the unsupported SNP:
rsnps::ncbi_snp_query( c( "rs10921647", "rs10921645" ) )
# Getting info about the following rsIDs: rs10921647, rs10921645
# # A tibble: 2 × 16
# query  chrom…¹     bp class rsid  gene  alleles ances…² varia…³ seqname hgvs  assem…⁴ ref_seq
# <chr>  <chr>    <dbl> <chr> <chr> <chr> <chr>   <chr>   <chr>   <chr>   <chr> <chr>   <chr>  
#   1 rs109… 1       1.95e8 snv   rs10… ""    G,A,T   G       A,T     NC_000… NC_0… GRCh38… G      
# 2 rs109… 1       1.95e8 snv   rs10… ""    C,A,G   C       A,G     NC_000… NC_0… GRCh38… NA     
# # … with 3 more variables: minor <chr>, maf <dbl>, maf_population <list>, and abbreviated
# #   variable names ¹​chromosome, ²​ancestral_allele, ³​variation_allele, ⁴​assembly
# # ℹ Use `colnames()` to see all variable names
```
  

**Suggested fix**

 `variant.response.content` contains a list named `unsupported_snapshot_data`. You could use this to make your code more flexible. For example by adding `& is.null(variant.response.content$unsupported_snapshot_data)` to the if statement on [line  275](https://github.com/ropensci/rsnps/blob/4f434c04d78d68a356a2e373f46755d43a72cb8e/R/ncbi_snp_api.R#L275), and adding a separate if statement that handles SNPs that are no longer supported. Something like this for example:

```r
## if snp is no longer supported by dnSNP
if (!is.null(variant.response.content$unsupported_snapshot_data)) {
  
  Produce warning and return an empty data frame
  
}
```

**Session Info: `devtools::session_info()` or `sessionInfo()`**

<details> <summary><strong>Session Info</strong></summary>

  
```r
R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8  LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] magrittr_2.0.3 dplyr_1.0.10  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9        knitr_1.41        feather_0.3.5     hms_1.1.2         tidyselect_1.2.0 
 [6] R6_2.5.1          rlang_1.0.6       fastmap_1.1.0     fansi_1.0.3       stringr_1.5.0    
[11] httr_1.4.4        plyr_1.8.8        rsnps_0.5.0.0     tools_4.2.2       DT_0.26          
[16] xfun_0.35         utf8_1.2.2        cli_3.4.1         htmltools_0.5.3   ellipsis_0.3.2   
[21] yaml_2.3.6        digest_0.6.30     tibble_3.1.8      httpcode_0.3.0    lifecycle_1.0.3  
[26] tzdb_0.3.0        readr_2.1.3       htmlwidgets_1.5.4 vctrs_0.5.1       curl_4.3.3       
[31] crul_1.3          glue_1.6.2        evaluate_0.18     rmarkdown_2.18    stringi_1.7.8    
[36] pillar_1.8.1      compiler_4.2.2    generics_0.1.3    jsonlite_1.8.3    pkgconfig_2.0.3  
```

**Additional context**
I am not aware of any simple and relatively fast way of identifying SNPs that are no longer supported and excluding them from `snps` prior to using `ncbi_snp_query()`.

**Comment by jooolia** [comment]
Thank you @ellisifnygaard for the bug report and the detailed information on how to reproduce. I will look into getting a fix out in the coming weeks. Many thanks, Julia

**Comment by jooolia** [comment]
Dear @ellisifnygaard  Thanks again for the detailed bug report. I have added in support for this case in a similar way to snps that have been withdrawn from dbsnp. This fix is now in the dev branch (merged in #162). I will aim to do a new cran release in early in the new year. 
Thanks! Julia
